spark-operator/src/main/java/org/apache/spark/k8s/operator/config/SparkOperatorConf.java

```diff
@@ -0,0 +1,427 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.spark.k8s.operator.config;
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import io.javaoperatorsdk.operator.api.config.LeaderElectionConfiguration;
+import io.javaoperatorsdk.operator.processing.event.rate.LinearRateLimiter;
+import io.javaoperatorsdk.operator.processing.event.rate.RateLimiter;
+import io.javaoperatorsdk.operator.processing.retry.GenericRetry;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+
+import org.apache.spark.k8s.operator.listeners.SparkAppStatusListener;
+import org.apache.spark.k8s.operator.reconciler.SparkReconcilerUtils;
+
+/** Spark Operator Configuration options. */
+@Slf4j
+public class SparkOperatorConf {
+  public static final ConfigOption<String> OperatorAppName =
+      ConfigOption.<String>builder()
+          .key("spark.kubernetes.operator.name")
+          .typeParameterClass(String.class)
+          .description("Name of the operator.")
+          .defaultValue("spark-kubernetes-operator")
+          .enableDynamicOverride(false)
+          .build();
+  public static final ConfigOption<String> OperatorNamespace =
+      ConfigOption.<String>builder()
+          .key("spark.kubernetes.operator.namespace")
+          .typeParameterClass(String.class)
+          .description("Namespace that operator is deployed within.")
+          .defaultValue("default")
+          .enableDynamicOverride(false)
+          .build();
+  public static final ConfigOption<String> OperatorWatchedNamespaces =
+      ConfigOption.<String>builder()
+          .key("spark.kubernetes.operator.watchedNamespaces")
+          .description(
+              "Comma-separated list of namespaces that the operator would be "
+                  + "watching for Spark resources. If unset, operator would "
+                  + "watch all namespaces by default.")
```

Comment:
Hmm. Is it scalable?
> watch all namespaces by default

Reply:
I believe so. When operating at cluster level, josdk would make only 1 watch at cluster level. This is also (I believe) most operators do.
From API server perspective - actually it might be happier with cluster mode. As far as I recall under the hood "namespaced" watch is actually applying one more filter on top of the cluster level resources, which might increase ETCD load compared with the cluster level one.